src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineViewModel.ViewModelStateDataChange.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.VisualStudio.Text.Editor;
+
+namespace Microsoft.VisualStudio.LanguageServices.DocumentOutline
+{
+    internal sealed partial class DocumentOutlineViewModel
+    {
+        private sealed record ViewModelStateDataChange(string? SearchText, CaretPosition? CaretPositionOfNodeToSelect, ExpansionOption? ExpansionOption, bool DataUpdated);
```

Comment:
doc what DataUpdated means.  Doc what one would expect from all of these value.  like is only one of thse ever set at a time?

3 of them are nullable, one is a bool.  i'm not sure i get the relationship.